jetty-cdi/src/main/java/org/eclipse/jetty/cdi/CdiServletContainerInitializer.java

```diff
@@ -47,11 +57,40 @@ public void onStartup(Set<Class<?>> c, ServletContext ctx)
         {
             ServletContextHandler context = ServletContextHandler.getServletContextHandler(ctx);
             Objects.requireNonNull(context);
-            context.getObjectFactory().addDecorator(new CdiDecorator(context));
-            context.setAttribute(CDI_INTEGRATION_ATTRIBUTE, "CdiDecorator");
-            LOG.info("CdiDecorator enabled in " + ctx);
+
+            // Test if CDI is in the webapp by trying to load the CDI class.
+            ClassLoader loader  = context.getClassLoader();
+            if (loader == null)
+                Loader.loadClass("javax.enterprise.inject.spi.CDI");
+            else
+                loader.loadClass("javax.enterprise.inject.spi.CDI");
```

Comment:
wow, these 2 loadClass calls are subtle in their difference. (`L` vs `l`)
I stared at them for far too long to understand why these existed like this.

Reply:
Indeed. I had thought `Loader` had a utility method to hide that if... but it didn't.  Might create one.

Reply:
I didn't create the utility method in the end... as it would not be clear if it was a search for the class (like we used to have that caused problems) or this explicit if.